src/component/plotly-graph/inject.js

```diff
@@ -53,6 +56,28 @@ function inject (opts = {}) {
   return parts
 }
 
+function resolvePlotlyJS (v) {
+  if (isUrl(v)) {
+    // save the plotly-bundle to tmpdir and use it:
+    const curlOpts = {
+      'url': v,
+      'retries': 10,
+      'output': path.join(os.tmpdir(), 'plotly-graph-exporter-build', 'plotly-bundle.js')
+    }
+
+    curl.request(curlOpts, function (err) {
+      if (err) {
+        throw new Error(`An error occured while retrieving PlotlyJS from: ${v}. Error: ${err}`)
+      }
+    })
+
+    return 'plotly-bundle.js'
```

Comment:
Hmm. So here you return a string before the request is finished. This could lead to a race condition where the browser window would open before the inject step is done fetching the plotly.js bundle. Sounds like we need to make the inject step async (just like parse, render and convert steps).

Reply:
It shouldn't be too hard to make the inject step async as [create-index.js](https://github.com/plotly/image-exporter/blob/1fae2a7efafe4d784a4dd996ec06956788b0ddaf/src/util/create-index.js) (i.e. where the component inject methods are called is already async.